src/main/java/org/jenkinsci/maven/plugins/hpi/TestInsertionMojo.java

```diff
@@ -102,22 +105,29 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             getLog().info("Skipping auto-test generation because we are targeting Jenkins "+target+" (at least 1.327 is required).");
             return;
         }
+        if (javaLevel != null && javaLevel.compareTo("7") < 0) {
+            getLog().info("Skipping auto-test generation because java.level= "+javaLevel+" in pom.xml (at least 7 is required).");
+            return;
+        }
 
         try {
             File f = new File(project.getBasedir(), "target/generated-test-sources/injected");
             f.mkdirs();
             File javaFile = new File(f, injectedTestName + ".java");
             PrintWriter w = new PrintWriter(new OutputStreamWriter(new FileOutputStream(javaFile)));
-            w.println("import java.util.*;");
             w.println("/**");
             w.println(" * Entry point to auto-generated tests (generated by maven-hpi-plugin).");
-            w.println(" * If this fails to compile, you are probably using Hudson &lt; 1.327. If so, disable");
-            w.println(" * this code generation by configuring maven-hpi-plugin to &lt;disabledTestInjection&gt;true&lt;/disabledTestInjection&gt;.");
+            w.println(" * It's possible to disable this code generation by configuring maven-hpi-plugin to {@code <disabledTestInjection>true</disabledTestInjection>}.");
             w.println(" */");
             w.println("public class " + injectedTestName + " extends junit.framework.TestCase {");
+            w.println("  /**");
+            w.println("   * Build test.");
+            w.println("   * @return test");
+            w.println("   * @throws Exception e");
+            w.println("   */");
             w.println("  public static junit.framework.Test suite() throws Exception {");
             w.println("    System.out.println(\"Running tests for \"+" + quote(project.getGroupId() + ":" + project.getArtifactId() + ":" + project.getVersion())+");");
-            w.println("    Map<String, Object> parameters = new HashMap<String, Object>();");
+            w.println("    java.util.Map<String, Object> parameters = new java.util.HashMap<>();");
```

Comment:
Is this really worth disabling Java 6 support for??

Reply:
I think that the jenkins plugins upgrading to a recent version of plugin parent and of maven hpi plugin can upgrade their java version to 1.7 at least. I see no reason they can't upgrade their java version at that moment: what are they waiting otherwise?
So my answer is yes, it's worth it.